docs/v3-ea-components/transport-types/http-transport.md

```diff
@@ -1,3 +1,135 @@
+
+#### Successful response structure
+*ProviderResult* is an object that contains two keys `params` and `response` . `params` is the set of parameters that uniquely relate to the response . `response` is the value that will be returned as response from External Adapter. `response` should contain `data` and `result` properties.
+
+Example response structure
+```typescript
+parseResponse: (params, response, config) => {  
+    return params.map((param) => {
+	  const value = response.data[param.symbol].price  
+      return {  
+        params: param, // Note that this should be single input parameter, not the params array above
+        response: {  
+          data: {  
+            result: value,  
+          },  
+          result: value,  
+        },  
+      }  
+    })  
+  }
+```
+#### Error response structure
+It is also possible to store and return errors as well, in case Data Provider has not returned information that was expected.  In case of error, instead of `data` and `result`, the `resonse` object should contain `statusCode` for the errored response and `errorMessage` that will be sent back from the adapter.
+
+Refactored example that shows both error and success examples
+```typescript
+parseResponse: (params, response, config) => {  
+    return params.map((param) => {
+	  const value = response.data[param.symbol]?.price  
+	  if (!value) {
+		  // no value provided from DP. Sending error response back
+		  return {
+			   params: param,
+			   response: {
+				   errorMessage: `Could not retrieve valid data from Data Provider for symbol ${param.symbol}`,
+				   statusCode: 502
+			   }
+		  }
+	  }
+	  // if everything is ok we send success response
+      return {  
+        params: param, // Note that this should be single input parameter, not the params array above
+        response: {  
+          data: {  
+            result: value,  
+          },  
+          result: value,  
+        },  
+      }  
+    })  
+  }
+```
+### Type params of the transport
+`HttpTransport` accepts one type param that describes the request and response types for the endpoint and the transport. It is used to ensure that the transport and endpoint are compatible.
+```typescript
+const transport = new HttpTransport<EndpointTypes>({
+```
+Example structure of `EndpointTypes`
+```typescript
+export type EndpointTypes = {  
+ Request: {  
+	Params: CryptoRequestParams // type of input parameters
+  }  
+ Response: SingleNumberResultResponse // type of External Adapte response. `SingleNumberResultResponse` is built in type that indicates that both `data` and `result` are numbers
+ Settings: BaseAdapterSettings  // type of adapter config
+ Provider: {  
+  RequestBody: never  // type of request body in case of POST requests. `never` for GET requests
```

Comment:
You should technically be able to have a request body in a GET request, but usually not. Just a nit pick

```suggestion
  RequestBody: never  // type of request body in the case of POST requests. This is usually `never` for GET requests, and is *not* the same as query params
```